scripts/merge_parquet_files.py

```diff
@@ -4,23 +4,48 @@
 
 import pandas as pd
 
+try:
+    import pyarrow.parquet as pq
+except ImportError:
+    _HAS_PYARROW_ = False
+else:
+    _HAS_PYARROW_ = True
+
+
 def load_parquet_files_into_dataframe(parquet_files):
     return pd.concat(
         [pd.read_parquet(f) for f in parquet_files],
         axis=0,
         ignore_index=True,
     )
 
-def run(input_files, output_file, sort_input_files=False, parquet_engine='pyarrow'):
+
+def run(input_files, output_file, sort_input_files=False,
+        parquet_engine='pyarrow', assume_consistent_schema=False):
     if sort_input_files:
         input_files = sorted(input_files)
-    df = load_parquet_files_into_dataframe(input_files)
-    df.to_parquet(
-        output_file,
-        engine=parquet_engine,
-        compression=None,
-        index=False,
-    )
+
+    if assume_consistent_schema:
+        if parquet_engine != "pyarrow" or not _HAS_PYARROW_:
+            raise ValueError("Must use/have pyarrow when assume_consistent_schema is set to True")
+        if not input_files:
+            raise ValueError("No input files to merge")
+
+        t = pq.read_table(input_files[0])
+        with pq.ParquetWriter(output_file, t.schema, flavor='spark') as pqwriter:
```

Comment:
Great to see the flavor explicitly set! I recall we had some trouble in the past reading with Spark the raw parquet files generated by those scripts (especially on the way files are partitioned internally) - I hope this will enforce compatibility with a spark consumer.